### ISSUE TITLE
Property '$formatMessage' does not exist on type ...

### ISSUE THREAD

**Issue by Ricki-BumbleDev** [formatjs/formatjs#3691]
## Which package?
`vue-intl`

## Describe the bug
Using latest versions of vue, typescript and vue-intl, this gives a typescript error:
```html
<template>
  <h1>{{ $formatMessage({ id: 'test' }) }}</h1>
</template>
```
Error:
```
Property '$formatMessage' does not exist on type '{ $: ComponentInternalInstance; $data: {}; $props: Partial<{}> & Omit<Readonly<ExtractPropTypes<{ isRevealed: { type: BooleanConstructor; required: true; }; }>> & VNodeProps & AllowedComponentProps & ComponentCustomProps, never>; ... 10 more ...; $watch(source: string | Function, cb: Function, options?: WatchOptions...'.
```

### Stackblitz URL

https://stackblitz.com/edit/vue-intl-typing-issue?file=src%2FApp.vue,src%2Fmain.ts

Run `yarn type-check` to see the error.

## Expected behavior
No type errors.

## Screenshots
<img width="994" alt="image" src="https://user-images.githubusercontent.com/13829781/176318896-e2427d03-9765-4358-8911-38041b4f7738.png">

## Suggested Solution
Typing for global properties needed, see https://vuejs.org/guide/typescript/options-api.html#augmenting-global-properties and https://stackoverflow.com/a/64189003/5577580

```ts
import { IntlFormatters, IntlShape } from '@formatjs/intl';

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    $intl: IntlShape;
    $formatMessage: IntlFormatters['formatMessage'];
    $formatDate: IntlFormatters['formatDate'];
    $formatTime: IntlFormatters['formatTime'];
    $formatDateTimeRange: IntlFormatters['formatDateTimeRange'];
    $formatRelativeTime: IntlFormatters['formatRelativeTime'];
    $formatDisplayName: IntlFormatters['formatDisplayName'];
    $formatNumber: IntlFormatters['formatNumber'];
    $formatList: IntlFormatters['formatList'];
  }
}
```

I can probably create a PR for the fix if a maintainer is interested.

**Comment by longlho** [comment]
yup PR's always welcome

**Comment by longlho** [comment]
closed in #3703
